packages/ui/src/composites/wui-profile-button/index.ts

```diff
@@ -40,10 +42,10 @@ export class WuiProfileButton extends LitElement {
         <wui-flex gap="xs" alignItems="center">
           <wui-text variant="large-600" color="fg-100">
             ${UiHelperUtil.getTruncateString({
-              string: this.address,
-              charsStart: this.isProfileName ? 18 : 4,
-              charsEnd: this.isProfileName ? 0 : 4,
-              truncate: this.isProfileName ? 'end' : 'middle'
+              string: this.profileName ?? this.address,
```

Comment:
These should be handled in https://github.com/WalletConnect/web3modal/pull/2276/files